src/Handler/TagHandler.php

```diff
@@ -59,10 +87,48 @@ public function __construct(CacheInvalidator $invalidator, $tagsHeader = 'X-Cach
      */
     public function invalidateTags(array $tags)
     {
-        $tagExpression = sprintf('(%s)(,.+)?$', implode('|', array_map('preg_quote', $tags)));
+        $tagExpression = sprintf('(%s)(,.+)?$', implode('|', array_map('preg_quote', $this->escapeTags($tags))));
         $headers = array($this->tagsHeader => $tagExpression);
         $this->invalidator->invalidate($headers);
 
         return $this;
     }
+
+    /**
+     * Get the HTTP header name that will hold cache tags
+     *
+     * @return string
+     */
+    public function getTagsHeader()
+    {
+        return $this->tagsHeader;
+    }
+
+    /**
+     * Format the tags string.
+     *
+     * @param array $tags Non-empty list of tags to output
+     *
+     * @return string
+     */
+    protected function formatTagsString(array $tags)
+    {
+        return implode(',', $this->escapeTags(array_unique($tags)));
+    }
+
+    /**
+     * Make sure that the tags are valid.
+     *
+     * @param array $tags The tags to escape.
+     *
+     * @return array Sane tags.
+     */
+    protected function escapeTags(array $tags)
+    {
+        array_walk($tags, function (&$tag) {
+            $tag = str_replace(array(',', "\n"), array('_', '_'), $tag);
```

Comment:
we missed doing this before, leading to `,` in a tag breaking the tagging. not sure if we need \n or not